docs/dev-process/coding-guidelines.md

```diff
@@ -123,6 +123,15 @@ Some general guidelines:
   Instead, use `var example = obj as Example` or the C# 7 syntax `if (obj is Example example) {...}` as appropriate.
   In this way you can avoid converting to the type twice.
 
+* Use generic collections instead of `ArrayList` to avoid type casting and unnecessary boxing whenever possible.
```

Comment:
Maybe we should recommend use generic collections over classicc collections (not only `ArrayList`)?

Reply:
I mainly see `ArrayList` in past experiences. What other collection types do you refer to?

Reply:
I meant something like "Use generic collections _in particular_ instead of `ArrayList` to avoid type casting... "
It seems we use `Dictionary<TKey, TValue>`.

Reply:
I think I get your point. I will mention `Stack`, `Queue` and `Hashtable` along with `ArrayList`.